src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -85,6 +85,58 @@ class java_bytecode_convert_classt:public messaget
   static void add_array_types(symbol_tablet &symbol_table);
 };
 
+/// Auxiliary function to extract the generic superclass reference from the
+/// class signature. If the superclass is not generic, it returns an empty
+/// string.
+/// \param signature Signature of the class
+/// \return Reference of the generic superclass, or empty if the superclass
+/// is not generic
+static std::string
+extract_generic_superclass_reference(const std::string &signature)
```

Comment:
Consider using `optionalt<std::string>` rather than the empty string to denote the non-generic case.